### ISSUE TITLE
Always show the full screen button

### ISSUE THREAD

**Issue by nbmrjuhneibkr** [TeamNewPipe/NewPipe#4478]
### Checklist

- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
In v0.20, full screen button is hidden when automatic screen rotation is enabled in the OS.  
The button should always be accessible, regardless of the system settings.

#### Is your feature request related to a problem? Please describe it
When the device lies on any flat surface and auto-rotate is enabled in the OS, there is no way to play videos in full screen mode without picking the device up to trigger the sensor, or disabling auto-rotate.

#### Additional context
A user may want to watch videos without picking the device up. Disabling auto-rotation globally is not an acceptable solution.

#### How will you/everyone benefit from this feature?
When combined with the manual video rotation control (which should also be restored), this will restore all manual controls that were available in NewPipe before the 0.20 update.

**Comment by avently** [comment]
I disagree with the issue. Your example with flat surface is like "I need to say something related to the issue". 
I can agree with some of your arguments related to fullscreen button in case of locked orientation but not unlocked.
Whenever I see disappeared fullscreen button it is a sign of enabled autorotation. And I'm 100% sure that I can rotate the device and get whatever orientation I want. With always shown button I need to use it or always look at status bar's panel (which is not an option on android 4.4 because the autorotation button isn't present there). This is something I know usefulness of.

**Comment by nbmrjuhneibkr** [comment]
I disagree with your disagreement. Your responses often look like "I need to find a reason to dismiss this issue".

Sometimes I keep my phone on a table and play YouTube videos on it while doing work. I didn't have to pick up the phone to change the orientation before v0.20.
Please stop forcing your personal usage scenarios on everyone else.

>With always shown button I need to use it or always look at status bar's panel (which is not an option on android 4.4 because the autorotation button isn't present there).

Then why not have two slightly different icons? This is an easy fix.

**Comment by avently** [comment]
@nbmrjuhneibkr 
>Please stop forcing your personal usage scenarios on everyone else.

Please stop saying what I should do and I will not say where you need to go.

**Comment by nbmrjuhneibkr** [comment]
I'm only mentioning things that have direct impact on the app and the way people use it.

And it looks like you're one step away from taking the route of personal attacks.

Anyway, back to the issue. If someone is going implement this while addressing "how do I know if auto-rotate is enabled" thing, my idea would be to have a normal full screen button when it's disabled, and the button with added circular arrows either on the outside or on the inside when it's enabled.

**Comment by avently** [comment]
@nbmrjuhneibkr 
> And it looks like you're one step away from taking the route of personal attacks.

It looks like you like to call things in other words which aren't near truth. Even now you provoking.

I'm not interested in talking about people when developing the app. It's counterproductive. Instead I prefer to talk about real app's issues. So if you like this `correct` route we can follow it.

**Comment by nbmrjuhneibkr** [comment]
While @avently was busy writing a comment about the interest (or lack thereof) in talking about people, I came up with an easier solution for https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-706611050  
Instead of designing new icon for the full screen button, just change the color of the button when auto-rotation is enabled. Simple.

**Comment by hipunk** [comment]
"I", "I", "I"... Does this guy talk about anything else than himself or what he thinks?
Isn't this a community project of more than 500 contributors? Why is avently allowed to remove so many features when he's this clueless about smartphones? (e.g. #4414: never heard of GPRS. Doesn't know multiple WiFis might have different throughput. Thinks android phones have tablet UIs. Confuses accessibility settings with build.prop DPI flags. Thinks every device in the world is held using hands and can be rotated. Makes fun of the concept of reducing traffic, although it increases battery runtime like almost nothing else in a smartphone)

How is one single person allowed to introduce so much breakage?
Let me rephrase: Avently is deleting functional, proven and real world tested work of others! Stop him already! 0.20.0 is unusable, yes, useless, on a Samsung Galaxy S4. He said we can blame him because he is responsible. Here is the blaming, stop breaking stuff for no gains, avently.

**Comment by avently** [comment]
@hipunk how old are you?

**Comment by hipunk** [comment]
> @hipunk how old are you?

Your code sucks. Bitch, eat shit and die.

**Comment by avently** [comment]
>>@hipunk how old are you?

>Your code sucks. Bitch, eat shit and die.

Thanks, I know more than needed :stuck_out_tongue_closed_eyes:

This provocateur is done.

**Comment by hipunk** [comment]
Sorry, that was too harsh from me. However... you know. I mean it, but not in a rude way.

Have fun! That's all that matters.

**Comment by nbmrjuhneibkr** [comment]
@avently 
>how old are you?
>This provocateur is done.

I thought we were going to stop talking about people here...

Here's another reason for implementing this (or rather fixing, since this wasn't an issue before 0.20), if I haven't given enough reasons already: feature parity with the official YouTube app. This feature is present there, probably for the same exact reason as the one I mentioned. NewPipe losing features that are present in the official client is not a step in the right direction.

**Comment by avently** [comment]
@nbmrjuhneibkr 
>feature parity with the official YouTube app

I'm not interested in feature-parity with YouTube app. This can't be a goal. YouTube app doing something great and something bad. In this case showing a button to rotate orientation when it's unlocked globally is bad behavior because if a user chose to autorotate his screen's orientation it means he likes this behaviour. Forcing rotation by a button it's a step in a wrong direction in this case.

**Comment by nbmrjuhneibkr** [comment]
>YouTube app doing something great and something bad.
This feature adds more choice and doesn't break anything. Definitely not a "bad" feature. Better than some that have been implemented, arguably.
>if a user chose to autorotate his screen's orientation it means he likes this behaviour
As I already said, _auto-rotation doesn't work in a use case described in this issue_, but user may still want to keep it enabled globally _for other use cases_. There should be a manual override for this _in the app_.
>Forcing rotation by a button it's a step in a wrong direction in this case.
The app isn't forcing anything, _the user does_. This is user choice that doesn't exist anymore in v0.20.

**Comment by stnby** [comment]
Omg please keep the full screen button rotating screen doesn't even put it in full screen this is so annoying.

**Comment by opusforlife2** [comment]
Reply to https://github.com/TeamNewPipe/NewPipe/issues/4500#issuecomment-709545368

@avently I suspect there may be a strong overlap between users who want the full screen button even with unlocked rotation, and users who want automatic full screen on tapping the thumbnail. So we should wait for more information.

**Comment by avently** [comment]
@opusforlife2 what information you want to get?

**Comment by opusforlife2** [comment]
If everyone involved is satisfied with automatic full screen behaviour then we won't need this feature anyway. If not, it could be discussed further.

**Comment by nbmrjuhneibkr** [comment]
I don't see how automatic full screen player could be a replacement for this feature request.  

https://github.com/TeamNewPipe/NewPipe/issues/4152 aims to restore (as an option, I think) v0.18 behavior where videos played *only* in full screen, while this issue is about expanding *current behavior* of the player by displaying the full screen button (perhaps a slightly different looking - https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-706626385 ) when screen rotation is enabled in the OS.

**Comment by nahuhh** [comment]
Gotta agree with op..
Need to force rotation.

I want my videos to STAY in landscape, not flip to portrait when I move 

Seems like it was more work than it was worth to make the icon hide based on the device orientation toggle..

Thanks. Been using the app for years and it's great. The new ui looks good and I like the direction.  But it's not usable. 

Background play and current queue management is pretty much gone? (Compared to older versions) 
Old version was a solid "1.1"
This feels like a super alpha pre2.0 release.

**Comment by nbmrjuhneibkr** [comment]
> force rotation

That's a different issue: https://github.com/TeamNewPipe/NewPipe/issues/4500

And maybe also related to this, depending on personal preference: https://github.com/TeamNewPipe/NewPipe/issues/4152

This one is about the full screen button (which doesn't necessarily lead to video rotation) currently being hidden when auto-rotation is enabled in the OS.

**Comment by nbmrjuhneibkr** [comment]
> After all I think that having the choice is the important thing here

I feel that removing as much choice as possible and leaving users with only one "correct" way to use NewPipe was one of the main goals of v0.20 development, sadly.

**Comment by mhmdanas** [comment]
I want to work on this, but it's been suggested that I should ask for people's opinions through a poll first, so here I am. Vote by reacting with the emoji beside the option you vote for (e.g. :tada: for 2).

1. :+1: Always show the full screen button (without option to disable this)
2. :tada: Hide when auto-rotation is enabled, which is the current behavior (also without an option to disable this)
3. Add an option to choose between the two behaviors
    1. :heart: Make 1 the default option
    2. :eyes: Make 2 the default option

Somewhat related: @Redirion suggested creating a "fullscreen control" setting, which in addition to the previous options has an "always play videos in fullscreen" option. The only issue I see with this idea though is that that option is not mutually exclusive with the others (e.g. if you leave fullscreen while auto-rotation is enabled, which behavior do we assume?).

@TobiGr @opusforlife2 @B0pol @TiA4f8R @vkay94 @TacoTheDank @XiangRongLin @MD77MD @s1awek @shivasagarrao @wilzbach @T5000 @Bruceforce @ShareASmile @89z @thinsoldier @atmosfar @SameenAhnaf

**Note**: if you don't want to be mentioned by polls in the future, please react with :rocket:.

**Comment by 89z** [comment]
STOP SPAMMING THE THREAD. this is the third time I've gotten an email for a dumb vote request.

Just make a comment and people will see it if they are subscribed. Don't at mention every single person in the thread.

**Comment by Bruceforce** [comment]
> STOP SPAMMING THE THREAD. this is the third time I've gotten an email for a dumb vote request.
> 
> Just make a comment and people will see it if they are subscribed. Don't at mention every single person in the thread.

Why don't you just react with the  🚀 emoji then like suggested in the notes? At least you currently have not done so in this thread.

**Comment by mhmdanas** [comment]
@89z I apologize for this. Although I believe that I gave the option to react with :rocket: to not get mentioned in the future, I understand your frustration. Again, I'm sorry.

**Comment by avently** [comment]
@mhmdanas i don't think it's a good idea to base developer's decision on a poll. people that like how things work are not on github issues. so you'll get wrong results just because 99% of users are never be here and they like current behavior

**Comment by SameenAhnaf** [comment]
@avently If I am not wrong, polls need to be properly translated to get everyone's opinions. Only English users don't use it. Only Polish users don't use it. I wonder if that is possible as new users of unpopular languages are also coming.

Many behaviours like "Unified player", "Directly enter full screen", "Directly exit full screen" etc. were current behaviours of the past and ditched without users' opinions. At least, a good alternative should have been introduced. We never know others' needs. Developers will be able to give proper alternatives at least in most cases.

If any user misunderstands a topic or hardly uses any feature, that will create inaccuracies in poll results. Many users may not want to attend polls at all.

Most developers are at least conscious about the app and give proper statements. So, taking only developers' opinions is not a bad idea, I think. At least, they know what they are doing.

If you still want to add polls for users, that could be done too. Three more options in "Polls" section could be added in settings, "Notify me for all (English available)", "Notify me for selected languages" and "Notify me for selected settings". 

Still, developers-only polls could be added to get an estimated result. As expected, all users may not be able to attend polls on the same day. But developers will be able to give a quick results preview especially on complicated codes.

@mhmdanas I don't know if it is a proper place to ask. But I would like to get back "Directly enter full screen" and "Directly exit from videos". Related issues #5645 and #5675

In #5645, selecting "Always, on full screen" will let us "Directly enter full screen". In #5675, selecting "None" will let us "Directly exit full screen".

**Comment by mhmdanas** [comment]
@avently I'm aware of that, but it's at least better than going with one's own opinion only.

**Comment by avently** [comment]
@mhmdanas probably nobody likes how whatsapp works: account per device, inability to "easy" restore messages into another device, phone number-only connection with other users. But it's the most used messenger. Same with devices from Apple.
I mean in some situations users can share their use-case you unaware of, it helps to find a solution for their problems.
But in this situation it's counterproductive. Autorotation is an OS mechanism and it's a preferred way to use it like used here: if autorotation enabled hide fullscreen button otherwise show.

**Comment by SameenAhnaf** [comment]
@avently Or, probably, you don't understand how much time and efforts  are applied for each feature. Why are your imposing your decision on everyone else? This app is not your personal property that you can ditch whatever you don't like. You never know others' needs.

You want everyone's opinions? There are polls. We can notify the users.
>If you still want to add polls for users, that could be done too. Three more options in "Polls" section could be added in settings, "Notify me for all (English available)", "Notify me for selected languages" and "Notify me for selected settings".

And, yes, don't forget that Newpipe is an open source app. It is not a copyrighted app like WhatsApp. Every developer utilizes their precious time with no income.


>Autorotation is an OS mechanism and it's a preferred way to use it like used here: if autorotation enabled hide fullscreen button otherwise show.

You have already had your answer, dude. I repeat, you never know others' needs. Nobody is always supposed to put their phone on flat service. Why can't we solve a problem if it has a solution?
>When the device lies on any flat surface and auto-rotate is enabled in the OS, there is no way to play videos in full screen mode without picking the device up to trigger the sensor, or disabling auto-rotate.

**Comment by avently** [comment]
@SameenAhnaf,
>Or, probably, you don't understand how much time and efforts are applied for each feature

Same applies to you related to features I introduced. But no one cares, right? Right.

Many months are gone since my PRs. And as you can see there are no activity in a field of my changes. I mean developers probably like how things work. If the situation is not changed I think that most of the people are fine with the changes too. Because when I don't like something I'm trying to make a feature that solves my problem.

>Nobody is always supposed to put their phone on flat service. Why can't we solve a problem if it has a solution?

The solution is to rotate the phone. You don't even need to touch a screen. Fast and easy. And take a look at the first my post in the issue where I talked about objectives of my decision. If you don't see the problem I mentioned there as a problem I absolutely sure that you don't care about opinions of other people about which you blame me. If you understand why I did this you probably will change you statements

**Comment by SameenAhnaf** [comment]
@avently Do you think ditching a feature solves the problem? That's what I am talking about.

You at least know programming. You can fork the app at least. Users like me are waiting for months without any progress. I have already opened over 30 issue. But, again, ditching a feature will only make the situation worse.

This option is still necessary if you lie on the bed. We all know screen rotation doesn't properly work then. We need to put the phone in desired mode, pull down notification panel turn off screen rotation, close notification panel and then again need to click on screen rotation button. (Old phones only support portrait mode if screen rotation is off). So, showing the full screen button all the time is still necessary.

I don't know why it seems to be a waste of time or something for you. It will only appear if you are working on video interface. It should not be too obstructive.

**Comment by avently** [comment]
>You at least know programming. You can fork the app at least

so you suggest me to give what you want but make my life harder by supporting a fork? and that's because i have learned programming for a half of my life and you don't? what did i just read?

>We all know screen rotation doesn't properly work then. We need to put the phone in desired mode, pull down notification panel turn off screen rotation

this is how autorotation works. it's not about the app at all. With disabled autorotation you have the ability to see the button. so no problem here.

also. if you want you can install apps like MacroDroid and setup a macro which disables autorotation for certain apps and enables it ones you leave them. i used this method many years

**Comment by nbmrjuhneibkr** [comment]
>if you want you can install apps like MacroDroid and setup a macro which disables autorotation for certain apps and enables it ones you leave them. i used this method many years

So this is where we're at now? Suggesting third-party apps as a workaround to restore missing features that already existed in NewPipe before the last major release?

**Comment by Redirion** [comment]
can we please talk objectively about it? I like the implementation that the fullscreen button is not shown when auto rotation is enabled. Less screen cluttering and natural behaviour.

That's why I voted for :eyes: . Keep the current behaviour and add an option for everyone that needs the fullscreen toggle always visible.

**Comment by Stypox** [comment]
Note: this has to be discussed along with #4500

**Comment by mof22** [comment]
PLEASE restore the full screen reproduction in portrait mode.
In previous versions I can view a landscape video in portrait mode with the subtitles outside the video frame.
Now subtitles occupy an important part of the video box and frequently obstruct a relevant part of the image.
This is very useful for those who need the assistance of the subtitles when we do not understand the language of the video.
In my case I do not understand English when I hear it, but I can read it with some fluidity.

**Comment by nbmrjuhneibkr** [comment]
I think something needs to be clarified here: full-screen playback in portrait mode would require not only this issue to be resolved, but also https://github.com/TeamNewPipe/NewPipe/issues/4500 (which is more important).

**Comment by iamthesenate1** [comment]
> The solution is to rotate the phone. You don't even need to touch a screen. Fast and easy. 
Yeah in theory it is. But some older phones have sh*t auto-rotation that sometime works, and sometimes it doesn't (sometimes it works in some apps and sometime in others it doesn't,etc). If it stops working, android will report it as enabled but the video won't go full screen. So you have to stop it from settings and than press the button. When you swich applications you have to go (again) into the settings and enable it. It is very annoying. 
Newpipe was designed to support older devices too, right?

**Comment by piegamesde** [comment]
> The solution is to rotate the phone. You don't even need to touch a screen. Fast and easy.

It is. Except you need to know that this feature exists, and that rotating the phone is enough. You see, it is not discoverable. People want to tap the full screen button before turning their phone, and now they are confused because "the fullscreen mode is gone".

I'm only loosely following the whole debate, but it seems improvements to the discoverability of the feature have not been thoroughly discussed here yet. We shouldn't discuss "which of these two options is better", but instead focus on "which solution is the best". And the existing proposed options surely have room for improvement.

**Comment by avently** [comment]
@piegamesde 
>It is. Except you need to know that this feature exists, and that rotating the phone is enough. You see, it is not discoverable. People want to tap the full screen button before turning their phone, and now they are confused because "the fullscreen mode is gone".

Autorotation thing works across all apps. In order to see an image in landscape in a gallery a user rotates his phone. If you want to see a chart in fullscreen you do the same. If someone is unaware of system-wide behavior it's a problem of the user, not implementation.

I talk with many people and know that many of them doesn't even know about possibly to copy and paste a text with long click. They don't know that long click is exists at ail (and even copy and paste). It's a lack of knowledge not the problem of OS or apps.

I made long tap on arrow button that scrolls down to comments section. It's totally undiscoverable and non-intuitive but the autorotation thing is intuitive.

**Comment by avently** [comment]
@piegamesde 
> Yeah in theory it is. But some older phones have sh*t auto-rotation that sometime works, and sometimes it doesn't (sometimes it works in some apps and sometime in others it doesn't,etc)

I had many phones and they didn't have such problem. So you think we all have to deal with unneeded buttons with enabled autorotation because you have a broken phone? It's unfair, isn't it? There is a NewPipe legacy available, you can try it.

**Comment by iamthesenate1** [comment]
I'm not saying it should be the default option. But it should be an option. Newpipe legacy breakes agagain and again. Plus new features should be available to as many people possible. 
Open source is in my opinion abaut freedom and **choice**. Two older phones I have (Android 5 and 6) have this problem.
Indeed al the phones from 2017 and later I have work perfectly. This is why I voted for not showing the button by default.
There are people who stil use android 4.1(see discussionin updating to Exoplayer 2.12).

Just becouse you don't need a feature doesn't mean that all people don't need it.

**Comment by piegamesde** [comment]
> Autorotation thing works across all apps.

Yeah, but applications changing a display mode based on rotation are rare. A landscape gallery is still the same as a portrait gallery, it has the same buttons and such. This is not the same as a full screen mode, or if the gallery started a slide show in landscape mode or something like that.

> If someone is unaware of system-wide behavior it's a problem of the user, not implementation.

I don't think this is a system-wide behavior, see above.

------

I won't discuss the other points here, because they are fundamental yet tangential to this specific issue. I just want you to know that people come to me with "NewPipe has a bug, I can't fullscreen anymore", and sure you can blame them instead of improving the UX, but they'll have a bad day using the software anyways. Dismissing other people's problems does not make them go away.

**Comment by mof22** [comment]
@iamthesenate1
I agree that it should be optional.
In my home there are two old phones, one with Android 4.4 and another with Android 5.1.
At the moment I do not intend to change them. In part because it would be an expense inappropriate of money, and partly because I do not need anything more modern.
I am a person who uses the phone only to talk and send messages, and eventually for entertainment. For everything else I use a desktop PC.

**Comment by avently** [comment]
@piegamesde but even youtube app has the same behavior: when you rotate the phone to landscape, the screen opens in fullscreen. 
People often asks me how to install an app. They don't know about google play and where they can find apps... So, this problem should be solved by viewing videos and reading more about how phones work.

If problems like this one will be fixed by adding additional controls we ended up with the app full of buttons and text descriptions for them.

**Comment by nbmrjuhneibkr** [comment]
>If problems like this one will be fixed by adding additional controls we ended up with the app full of buttons and text descriptions for them.

_The horror._ An app that gives users options instead of forcing them to always use it in one specific way. People should be ashamed for wanting such thing.

I have already described a real use case that is no longer possible with v0.20 (unless auto-rotation is disabled globally in the OS): https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-706617217

**Comment by SameenAhnaf** [comment]
How about if we ditch full screen button completely and replace it with "Swipe up and reach to interface gesture" suggested in #5299? The full screen should stick on full screen even if auto rotation is on. As expected, auto rotation doesn't work properly while lying on bed.

No need to keep minimize button either. Back button is there to go to main player.

**Comment by SameenAhnaf** [comment]
#5459 is there for easy entrance. So, senior users should not fall in trouble while getting into full screen.

**Comment by SameenAhnaf** [comment]
How about "Swipe up and reach to interface gesture" for full screen suggested in #5299? This will be a 2 in 1!

If system rotation is on: Applying this gesture will lock the full screen from getting into main player. Back button is there to get into main player. Rotating screen will switch to main player or full screen as usual.
If system rotation is off: Get into full screen in an easier way without showing interface. Minimize and back button will work as usual.

**Comment by SameenAhnaf** [comment]
How about "Swipe up and reach to interface gesture" for full screen suggested in #5299? This will be a 2 in 1!

If system rotation is on: Applying this gesture will lock the full screen from getting into main player. Back button is there to get into main player. Rotating screen will switch to main player or full screen as usual if this gesture was not applied.
If system rotation is off: Get into full screen in an easier way without showing interface from main player. Back button will work to get into main player.

Yes, minimize or full screen button could be permanently ditched as it is no use anymore.

**Comment by mhmdanas** [comment]
> Yes, minimize or full screen button could be permanently ditched as it is no use anymore.

No, they can't. Android TV users can't use gestures.

**Comment by SameenAhnaf** [comment]
@mhmdanas I am not sure how Android TVs work now. But it's not necessary for smartphones and tablets anymore.

Plus, #5459 is there for direct entrance to full screen. So, senior users should not fall in problem.

**Comment by Wicoler** [comment]
We need this

**Comment by hellyeahmfs** [comment]
> Yeah this needs to be implemented

**Comment by mhmdanas** [comment]
People, PLEASE stop posting comments that just ask for this to be added. If you have no value to add to the discussion, don't make a comment. Just do a +1 reaction.

**Comment by tguen** [comment]
Let me repeat a bit from my duplicate report to explain why this feature is bad UX.

1. I have enabled the "sensors off" quick tile from developer settings, which disables the gyro sensor. This effectively disables auto-rotate without the app detecting that it's disabled, so simply detecting when it's enabled is inadequate. The app doesn't explain why the button hidden. I was frustrated for about 2 weeks because it disappeared and I couldn't figure out why. I downgraded the app and it was still missing. I only figured out why when I came here to report it as a bug and found a comment on another report that explains why it disappeared.

2. Suppose I want to watch a video in fullscreen while laying on my side. If I enable auto-rotate while laying down, it doesn't switch to fullscreen. If I rotate to landscape, then disable auto-rotate, it automatically exits fullscreen to switch to portrait mode.

After 6 months of voting, it seems only 1 of 12 people think it should stay the way it is. (That one person clearly has a toxic attitude that should make you consider whether you want them involved with your project.) Due to the above issues, I think the way it is shouldn't even be the default option even if the app explains the feature to the user so they are at least aware that they can dig through the settings menu for a way to show a button needed to resolve them. It's too big of a problem to justify hiding the button for a "cleaner UI".

**Comment by T5000** [comment]
Thank you for the write-up, @tguen. I have to say I have been following this for over 6 months now and I completely agree with everything you have written. The toxic attitude is just bad and should not be tolerated.

**Comment by opusforlife2** [comment]
@LeonTheOriginal Long-press on the top-right arrow.

**Comment by LeonTheOriginal** [comment]
@opusforlife2 Thanks for the info. I didn't know that there is a long press action for that botton which is pretty counterintuitive for an icon to open/close a submenu.

**Comment by qol3m** [comment]
I need this, too. 
Whether I disabled the sensors, 
whether gravity does not align with my viewing angle (surely you have been on your side in bed before??),
whether I turned rotate off so that it does not rotate other apps when i switch,
or whether the sensor is broken/flawed,

I have been missing this many times.

**Comment by yashpalgoyal1304** [comment]
ping

**Comment by ccuser44** [comment]
THIS IS AN ABSOLUTE MUST!

Its horribly annoying that this is not a feature. Furthermore if you have the phones gyroscope disabled the full screen button wont show at all

**Comment by Stypox** [comment]
@ccuser44 could you verify that #9693 always shows the fullscreen button, since you don't have an accelerometer for rotation?

**Comment by yashpalgoyal1304** [comment]
as there's still lack of consensus, how about taking a leap of faith... and shipping this implemented anyhow...

* with lack of it, those who want it are saying here.

* if the new behaviour causes disturbance, then those ppl will say too (like they did in v0.20 release and all other times too)

that will settle it automatically... rather than just hypothesizing about the percieved dislike of the feature, how about testing in wild now....


and... i am not saying to do wild burning things in this manner, this is pretty small harmless change, so, this testing by releasing and looking for public reaction is feasible here.

**Comment by yashpalgoyal1304** [comment]
_self hiding this as offtopic_

----
> Vote by reacting with the emoji
\- @ mhmdanas at https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-804659342

@drishal you have selectively voted on all 3 in a single vote poll: 👍 ❤️ 👀

pretty _smart_ eh.. 😏😃

**Comment by adamency** [comment]
> I disagree with the issue. I can agree with some of your arguments related to fullscreen button in case of locked orientation but not unlocked. Whenever I see disappeared fullscreen button it is a sign of enabled autorotation. With always shown button I need to use it or always look at status bar's panel (which is not an option on android 4.4 because the autorotation button isn't present there).

You don't seem to understand that this is "useful" (actually not really at all, since clicking on the full-screen button and rotating the device achieves exactly the same result, so you don't gain anything) to you simply because you're a developer of the program and know about this **undocumented** behavior. Most users don't have years of NewPipe use nor programming knowledge. So for the global Newpipe population, being misled into thinking the full screen button has suddenly disappeared and losing time before realizing they can just rotate" is happening much much more than what you're describing.

**Comment by oxou** [comment]
And this is why open source sucks in terms of discussions, countless wars and arguments over petty things, the time it has taken for everyone to yell back and forth could've been used to implement said feature.  Once again the open source community has shown their immeasurable incompetence.

**Comment by piegamesde** [comment]
@oxou If you have nothing constructive to add to the discussion, maybe consider not saying anything. Take your bad takes back to hacker news or wherever.

**Comment by AudricV** [comment]
> Once again the open source community has shown their immeasurable incompetence.

No, because we will implement the first option, before the player rewrite ([the one which won this poll](https://github.com/TeamNewPipe/NewPipe/issues/4478#issuecomment-804659342) and the easiest one to implement).

Thanks for being agressive against contributors and team members who work on this project in their free time.
